kernel/src/hil/text_screen.rs

```diff
@@ -18,7 +18,11 @@ pub trait TextScreen<'a> {
     /// Return values:
     /// - `SUCCESS`: The write command is valid and will be sent to the driver.
     /// - `EBUSY`: The driver is busy with another command.
-    fn print(&self, buffer: &'static mut [u8], len: usize) -> ReturnCode;
+    fn print(
+        &self,
+        buffer: &'static mut [u8],
+        len: usize,
+    ) -> Result<(), (ReturnCode, &'static mut [u8])>;
```

Comment:
Now that we have `ErrorCode`, if a HIL is going to return a `Result<>`, it makes more sense to return a `Result<(), ErrorCode>` than a `Result<(), ReturnCode>`.

In the long term the goal is to move all error handling to the former, and deprecate `ReturnCode`.

Reply:
If `ReturnCode` will be deprecated, functions HIL should return `Result<_, ErrorCode>`?

Reply:
That is the goal, though we are not enforcing it at the 2.0 switch because it will be *a lot* of refactoring work. But generally it does not make sense that it could be possible to return a `Success` variant in an `Err()`, so `ErrorCode` makes more sense than `ReturnCode` when used in a `Result`

Reply:
I'd rather have this merged, as we do need to rewrite in a similar way the screen driver and it is easier to have both of them rewritten at the same time.

Reply:
ok, that's fine with me.